data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/ListToMapProcessorConfig.java

```diff
@@ -41,35 +42,55 @@ static FlattenedElement fromOptionValue(final String option) {
     @NotEmpty
     @NotNull
     @JsonProperty("source")
+    @JsonPropertyDescription("The list of objects with `key` fields to be converted into keys for the generated map.")
     private String source;
 
     @JsonProperty("target")
+    @JsonPropertyDescription("The target for the generated map. When not specified, the generated map will be " +
+            "placed in the root node.")
     private String target = null;
 
     @JsonProperty("key")
+    @JsonPropertyDescription("The key of the fields to be extracted as keys in the generated mappings. Must be " +
+            "specified if `use_source_key` is `false`.")
     private String key;
 
     @JsonProperty("value_key")
+    @JsonPropertyDescription("When specified, values given a `value_key` in objects contained in the source list " +
```

Comment:
Can we use multi line strings now? We already migrated to java 17 right?

Reply:
Thought about it, but is it going to help here? The descriptions are often just single sentences (no line breaks in them).